paperqa/contrib/zotero.py

```diff
@@ -216,6 +216,81 @@ def iterate(
             num_remaining = limit - actual_i
 
         self.logger.info("Finished downloading papers. Now creating Docs object.")
+    
+    def iterate_collection(
+        self,
+        collection_name: str,
+        limit: int = 25,
+        start: int = 0,
+    ):
+        """Given a collection name, this will lazily iterate over papers in a Zotero library, downloading PDFs as needed.
+
+        Args:
+            collection_name (str): name of the collection in Zotero
+            limit (int, optional): the limit of papers to download. Defaults to 25.
+            start (int, optional): the start index of the papers to download. Defaults to 0.
+        """
+        # specfic collection
+        collections = self.collections()
+        collection_id = None
+
+        for collection in collections:
+            name = collection["data"]["name"]
+            if name  == collection_name:
+                collection_id = collection["data"]["key"]
+                break
+
+        if collection_id:
+            coll_items = self.collection_items(collection_id)
+            self.logger.info(f"Collection '{collection_name}' found: {len(coll_items)} items")
+                
+        else:
+            self.logger.info(f"Collection '{collection_name}' not found")
+            return
+        
+        max_limit = 100
```

Comment:
Maybe move everything here down into a separate function, to be called by both `iterate` and `iterate_collection`? The less reused code the better (for maintainability)

Reply:
I integrated `iterate_collection` to `iterate`, so that `iterate` can receive `collection_name` as one argument. 
details: #[bd881c2](https://github.com/whitead/paper-qa/pull/48/commits/bd881c27679137c73f5bea042b2448db4f4b155e)